internal/peer/lifecycle/chaincode/queryapproved.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright Hitachi America, Ltd. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package chaincode
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	lb "github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+	"github.com/hyperledger/fabric/bccsp"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// ApprovedQuerier holds the dependencies needed to query
+// the approved chaincode definition for the organization
+type ApprovedQuerier struct {
+	Command        *cobra.Command
+	EndorserClient EndorserClient
+	Input          *ApprovedQueryInput
+	Signer         Signer
+	Writer         io.Writer
+}
+
+type ApprovedQueryInput struct {
+	ChannelID    string
+	Name         string
+	Sequence     int64
+	OutputFormat string
+}
+
+// QueryApprovedCmd returns the cobra command for
+// querying the approved chaincode definition for the organization
+func QueryApprovedCmd(a *ApprovedQuerier, cryptoProvider bccsp.BCCSP) *cobra.Command {
+	chaincodeQueryApprovedCmd := &cobra.Command{
+		Use:   "queryapproved",
+		Short: fmt.Sprintf("Query the approved chaincode definition for my org."),
```

Comment:
This is no longer "for my org", it would be for any org, so perhaps more appropriately "Query an org's approved chaincode definition from its peer." or something like that.

Reply:
@jyellick Thank you for your review.
I will submit the updated patch to address the comments after #1262 is merged.